### ISSUE TITLE
Mechanism to merge multiple fields with folder structure

### ISSUE THREAD

**Issue by natevw** [natevw/putdoc#9]
Say I have a small document `{"_id":"example","type":"container", "name": "My Container", "owner":"me"}` that also needs attachments. Currently the only way to do so is:

```
example/
    _id		('example')
    type	('container')
    name	('My Container')
    owner	('me')
    _attachments
        file1.jpg
        file2.pdf
        [etc…]
```

That is, for every level in the object I have to make a binary choice to represent it entirely as a `*.json` object or entirely as files. And aside from the new `_docs`/`--docs` feature the top level of an object I don't even get a choice: all top-level fields have to have their own files.

Maybe we should make a certain field name like `_fields` or `__merge__` or similar that is expected to be an object that gets merged into the current level. Then instead of lots of little files, I could group some metadata within a JSON file:

```
helpful_feature/
    _id			('example')
    __merge__.json	('{"type":"container", "name": "My Container", "owner":"me"}')
    _attachments
        file1.jpg
        file2.pdf
        [etc…]
```

Note in this case I left `_id` as its own file as a stylistic choice; there's nothing special about it and I could have made an equivalent document by including it in the __merge__ object.

Other thoughts:

1. Need to review whether there's prior art here in any of the other similar utilities, and get feedback from authors of active projects to collaborate a solution that could become something of a standard if useful.
2. Also should consider whether to make this support *multiple* merge sources to facilitate say automated generation of certain fields or more logical/meaningful grouping.

**Comment by natevw** [comment]
Per the second point, maybe something like:

```
multimerge/
    _id			('example')
    __merge.meta.json	('{"type":"container", "name": "My Container", "owner":"me"}')
    __merge.git.json	('{"git-rev":"923f7fd", "git-branch": "master"}')
    _attachments
        file1.jpg
        file2.pdf
        [etc…]

```

It might be tempting to do something fancy like auto-prefixing (e.g. have it prepend `N-` or `N.` to all the properties in a file named "__merge.N.json") but I think that's a can of worms and (even assuming multiple merge files are useful) adding a whole new layer of extra semantics around them is **not** worth the complexity imo.

**Comment by natevw** [comment]
In my original examples I've used `__merge` with double-underscore thinking to "avoid conflict" with CouchDB's own single-underscore, but I don't think that's necessary — `_docs` is already single underscore and these fields don't go to CouchDB anyway.

Only concern might be w.r.t. behavior when such docs are processed with other tools that might not support this feature, but `__` vs. `_` wouldn't really help much on that front.

**Comment by natevw** [comment]
This is now in place, using `_data` as the special key name. Published as `putdoc@3.1.0`.
